src/treemodel/delta/transform.js

```diff
@@ -0,0 +1,490 @@
+/**
```

Comment:
I would split this file into 2: one which define the transformation API and one which adds all special cases. Also priorities might be moved to deltas as their properties.